### PR TITLE
ENH: Add AdditiveGaussianNoiseImageFilter Example

### PR DESCRIPTION
**Pull request by wilson-ben**
Adding an example utilizing the AdditiveGaussianNoiseImageFilter.

### CODE REVIEW

File: src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  if( argc != 5 )
+    {
+    std::cerr << "Usage: "<< std::endl;
+    std::cerr << argv[0];
+    std::cerr << " <InputFileName> <OutputFileName> [Mean] [Standard Deviation]";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
+    // Initialize and assign user provided variables
+  const char * inputImage = argv[1];
```
Comment (hjmjohnson) [inline src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: Can these be std::string?

File: src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  const char * outputImage = argv[2];
+  // get floating point numbers for the Mean and Standard Deviation to perform the algorithm
+  const double mean = atof(argv[3]);
```
Comment (hjmjohnson) [inline src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: Avoid atof. 
Prefer error checked std::stod over atof

Comment (wilson-ben) [reply src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: Will change, thanks

File: src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx (hunk 1 above)
Comment (wilson-ben) [reply src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: Yes, I will change them to std::string

Comment (jhlegarreta) [reply src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: @hjmjohnson @thewtex sorry to chime in but I'm wondering whether the same criterion is used in ITK. If not, and if it should be used, we may need to open an issue. Thanks.

File: src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  if( argc != 5 )
+    {
+    std::cerr << "Usage: "<< std::endl;
+    std::cerr << argv[0];
+    std::cerr << " <InputFileName> <OutputFileName> [Mean] [Standard Deviation]";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
+    // Initialize and assign user provided variables
+  const std::string * inputImage = argv[1];
```
Comment (hjmjohnson) [inline src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: @wilson-ben This code is now wrong.

`const std::string *`
`const std::string`

Comment (jhlegarreta) [reply src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx]: Change to `const char *` in 5c9b8e7.

**Comment by jhlegarreta** [comment]
@wilson-ben reviews are appreciated.

**Pull request merged**